standard/statements.md

```diff
@@ -1177,10 +1171,12 @@ The target of a `goto` *identifier* statement is the labeled statement with the
 >
 > *end note*
 
-The target of a `goto case` statement is the statement list in the immediately enclosing `switch` statement ([§12.8.3](statements.md#1283-the-switch-statement)) which contains a`case` label with the given constant value. If the `goto case` statement is not enclosed by a `switch` statement, if the *constant_expression* is not implicitly convertible ([§10.2](conversions.md#102-implicit-conversions)) to the governing type of the nearest enclosing `switch` statement, or if the nearest enclosing `switch` statement does not contain a `case` label with the given constant value, a compile-time error occurs.
+The target of a `goto case` statement is the statement list in the immediately enclosing `switch` statement ([§12.8.3](statements.md#1283-the-switch-statement)) which contains a `case` label with the given *constant_pattern*. If the `goto case` statement is not enclosed by a `switch` statement or if the nearest enclosing `switch` statement does not contain a `case` label with the given *constant_pattern*, a compile-time error occurs.
 
 The target of a `goto default` statement is the statement list in the immediately enclosing `switch` statement ([§12.8.3](statements.md#1283-the-switch-statement)), which contains a `default` label. If the `goto default` statement is not enclosed by a `switch` statement, or if the nearest enclosing `switch` statement does not contain a `default` label, a compile-time error occurs.
 
+It is a compile-time error for *constant_pattern* to target a *switch_label* containing a *case_guard*.
```

Comment:
This isn't right.  We need to define the target of the goto case so that it never matches one with a case guard.  See [this example](https://sharplab.io/#v2:CYLg1APgAgTAjAWAFBQMwAJboMLoN7LpGYZQAs6AsgBQCU+hxTAzgO4CWALgMYAW61VPQJImY4twCGzAKboMrXjIB26AGaSANrJCNx+9AG50AehPoArsoBOMyX0kAjTTL0H0U2fPSKV6rTpu7sTGZpY2dg7OrqLBHtJyqLqxwcAyGhaanMlxYgDmAPacBfFeqIZBxAC+bjUpJJgUlDDU7Mqc6AAewnV1QA==).